packages/shared/src/components/modals/SharedBookmarksModal.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { ReactElement } from 'react';
+import classNames from 'classnames';
+import { useMutation, useQuery, useQueryClient } from 'react-query';
+import request from 'graphql-request';
+import CopyIcon from '../../../icons/copy.svg';
+import { Button } from '../buttons/Button';
+import { ResponsiveModal } from './ResponsiveModal';
+import { ModalProps } from './StyledModal';
+import styles from './AccountDetailsModal.module.css';
+import { Switch } from '../fields/Switch';
+import {
+  BookmarksSharingData,
+  BOOKMARK_SHARING_MUTATION,
+  BOOKMARK_SHARING_QUERY,
+} from '../../graphql/bookmarksSharing';
+import { apiUrl } from '../../lib/config';
+import { TextField } from '../fields/TextField';
+import TwitterIcon from '../../../icons/twitter_color.svg';
+import SlackIcon from '../../../icons/slack.svg';
+import DiscordIcon from '../../../icons/discord_color.svg';
+import GithubIcon from '../../../icons/github.svg';
+import { ModalCloseButton } from './ModalCloseButton';
+import { useCopyLink } from '../../hooks/useCopyLink';
+
+export default function SharedBookmarksModal({
+  className,
+  ...props
+}: ModalProps): ReactElement {
+  const queryClient = useQueryClient();
+
+  const { data: bookmarksSharingData, isFetched } =
+    useQuery<BookmarksSharingData>('bookmarksSharing', () =>
+      request(`${apiUrl}/graphql`, BOOKMARK_SHARING_QUERY),
+    );
+
+  const { mutateAsync: updateBookmarksSharing } = useMutation<{
+    enabled: boolean;
+  }>(
+    () => {
+      const updatedValue = !bookmarksSharingData?.bookmarksSharing?.enabled;
+      return request(`${apiUrl}/graphql`, BOOKMARK_SHARING_MUTATION, {
+        enabled: updatedValue,
+      });
+    },
+    {
+      onSuccess: (data) => {
+        queryClient.setQueryData('bookmarksSharing', data);
+      },
+    },
+  );
+
+  const [isRssUrlCopied, copyRssUrl] = useCopyLink(
+    () => bookmarksSharingData?.bookmarksSharing?.rssUrl,
+  );
+
+  if (!isFetched || !bookmarksSharingData?.bookmarksSharing) {
+    return <></>;
+  }
+
+  return (
+    <>
```

Comment:
This empty fragment is not needed.
You can use `ResponsiveModal` as the direct parent

Reply:
@ayadotdev I see you resolved this but didn't apply my suggestion

Reply:
@idoshamun pushed now 👍